### ISSUE TITLE
polars.read_database fails on Oracle with  iter_batches=True. DPY-1006: cursor is not open

### ISSUE THREAD

**Issue by njesp** [pola-rs/polars#16206]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import oracledb
import polars as pl

sql = """
    select cast (rownum as number) id,
        cast (rownum as float) id2,
        cast ('textxxxxxx' as varchar2 (10 char)) v
    from (select rownum r
            from (select     rownum r
                        from dual
                    connect by rownum <= 1000) a,
                (select     rownum r
                        from dual
                    connect by rownum <= 1000) b,
                (select     rownum r
                        from dual
                    connect by rownum <= 1000) c
            where rownum <= 1000000000)
"""
oracledb.init_oracle_client()
con = oracledb.connect("@db")
d = pl.read_database(sql, con, iter_batches=True, batch_size=10000)
cnt = 0
for _d in d:
    cnt += _d.height
    print(cnt)
con.close()

```
LIne 24: for _d in d: gives this exception: oracledb.exceptions.InterfaceError: DPY-1006: cursor is not open

This is a description of the environment.
```
name: polars_poc
dependencies:
  - python=3.11.9
  - pip=24.0
  - pip:
    - oracledb==2.2.0
    - polars==0.20.25
```

### Issue description

This is in fact a continuation of issue [15470](https://github.com/pola-rs/polars/issues/15470)

### Expected behavior

It should run and print a line for each 10000 rows fetched. 

### Installed versions

<details>

```
--------Version info---------
Polars:               0.20.25
Index type:           UInt32
Platform:             Windows-10-10.0.19044-SP0
Python:               3.11.9 | packaged by Anaconda, Inc. | (main, Apr 19 2024, 16:40:41) [MSC v.1916 64 bit (AMD64)]

----Optional dependencies----
adbc_driver_manager:  <not installed>
cloudpickle:          <not installed>
connectorx:           <not installed>
deltalake:            <not installed>
fastexcel:            <not installed>
fsspec:               <not installed>
gevent:               <not installed>
hvplot:               <not installed>
matplotlib:           <not installed>
nest_asyncio:         <not installed>
numpy:                <not installed>
openpyxl:             <not installed>
pandas:               <not installed>
pyarrow:              <not installed>
pydantic:             <not installed>
pyiceberg:            <not installed>
pyxlsb:               <not installed>
sqlalchemy:           <not installed>
torch:                <not installed>
xlsx2csv:             <not installed>
xlsxwriter:           <not installed>
>>>
```

</details>

**Comment by alexander-beedie** [comment]
Hmm, I don't _think_ the two issues are actually related, but I'll look at this tonight (I've got an Oracle docker image at home to test on) 🤔👍

**Comment by alexander-beedie** [comment]
The above code works fine if you don't init the "thick" client (eg: omit the call to `oracledb.init_oracle_client()`). I haven't been able to set up a test env with the relevant client libraries to try in "thick" mode - endless errors so far... :-\

**Comment by alexander-beedie** [comment]
Got it; fix incoming... 👌

**Comment by alexander-beedie** [comment]
@njesp Fix is out - give it a go in the new `0.20.28` release.
